packages/discovery/src/peer-exchange/waku_peer_exchange_discovery.ts

```diff
@@ -198,6 +199,38 @@ export class PeerExchangeDiscovery
 
       const hasPeer = await this.components.peerStore.has(peerId);
       if (hasPeer) {
+        const { hasMaDiff, hasShardDiff } = await this.checkPeerInfoDiff(
+          peerId,
+          peerInfo,
+          shardInfo
+        );
+
+        if (hasMaDiff || hasShardDiff) {
+          log.info(
+            `Peer ${peerId.toString()} has updated multiaddrs or shardInfo, updating`
+          );
+          await this.components.peerStore.patch(peerId, {
+            ...(hasMaDiff && {
+              multiaddrs: peerInfo.multiaddrs
+            }),
+            ...(hasShardDiff &&
+              shardInfo && {
+                metadata: {
```

Comment:
will it replace previous value of `metadata` or will it combine them?

also, if transition happens from `shardInfo={...}` to `shardInfo=null` - should it be erased from `peerStore`?

Reply:
using `patch` will replace all existing metadata, might be better to use `merge` here